odrivecli.py

```diff
@@ -0,0 +1,1277 @@
+
+    agentProtocolServerPort = get_protocol_server_port(AGENT_PORT_REGISTRY_FILE_PATH)
+    desktopProtocolServerPort = get_protocol_server_port(DESKTOP_PORT_REGISTRY_FILE_PATH)
+
+    if not (agentProtocolServerPort or desktopProtocolServerPort):
+        print(REQUIRES_ODRIVE)
+        sys.exit(1)
+
+    if args.command == Authenticate.COMMAND_NAME:
+        command = Authenticate(agentPort=agentProtocolServerPort,
+                               desktopPort=desktopProtocolServerPort,
+                               authKey=getattr(args, Authenticate.AUTH_KEY_ARGUMENT_NAME))
+    elif args.command == Deauthorize.COMMAND_NAME:
+        command = Deauthorize(agentPort=agentProtocolServerPort, desktopPort=desktopProtocolServerPort)
+    elif args.command == Mount.COMMAND_NAME:
+        command = Mount(agentPort=agentProtocolServerPort,
+                        desktopPort=desktopProtocolServerPort,
+                        localPath=os.path.abspath(expand_user(getattr(args, Mount.LOCAL_PATH_ARGUMENT_NAME))),
+                        remotePath=getattr(args, Mount.REMOTE_PATH_ARGUMENT_NAME))
+    elif args.command == Unmount.COMMAND_NAME:
+        command = Unmount(agentPort=agentProtocolServerPort,
+                          desktopPort=desktopProtocolServerPort,
+                          localPath=os.path.abspath(expand_user(getattr(args, Unmount.LOCAL_PATH_ARGUMENT_NAME))))
+    elif args.command == Backup.COMMAND_NAME:
+        command = Backup(agentPort=agentProtocolServerPort,
+                         desktopPort=desktopProtocolServerPort,
+                         localPath=os.path.abspath(expand_user(getattr(args, Backup.LOCAL_PATH_ARGUMENT_NAME))),
+                         remotePath=getattr(args, Backup.REMOTE_PATH_ARGUMENT_NAME))
+    elif args.command == RemoveBackup.COMMAND_NAME:
+        command = RemoveBackup(agentPort=agentProtocolServerPort,
+                               desktopPort=desktopProtocolServerPort,
+                               localPath=os.path.abspath(expand_user(getattr(args,
+                                                                             RemoveBackup.LOCAL_PATH_ARGUMENT_NAME))))
+    elif args.command == Sync.COMMAND_NAME:
+        syncPath = os.path.abspath(expand_user(getattr(args, Sync.PLACEHOLDER_PATH_ARGUMENT_NAME)))
+        if getattr(args, RecursiveSync.COMMAND_NAME):
+            recSync = RecursiveSync(agentPort=agentProtocolServerPort,
+                                    desktopPort=desktopProtocolServerPort,
+                                    folderPath=syncPath,
+                                    noDownload=getattr(args, RecursiveSync.NO_DOWNLOAD_ARGUMENT_NAME))
+            recSync._sync_recursively()
+            print("Done with recursive sync of " + syncPath)
+            print("Current running downloads:")
+            command = DownloadsStatus(agentPort=agentProtocolServerPort, desktopPort=desktopProtocolServerPort)
```

Comment:
Why need to check for running downloads? After the recursive sync finishes, aren't all the placeholders replaced?